packages/client/src/pages/profile/edit-profile-dialog.tsx

```diff
@@ -0,0 +1,189 @@
+  IconButton,
+  Paper,
+  PropTypes,
+  TextField,
+} from "@material-ui/core";
+import styled from "styled-components";
+import CloseIcon from "@material-ui/icons/Close";
+import { useAppDispatch, useAppSelector } from "../../redux/store";
+import userService from "../../services/users";
+import { updateUser } from "../../redux/slices/user-slice";
+import { unwrapResult } from "@reduxjs/toolkit";
+import { useHistory } from "react-router-dom";
+import { PROFILE_PAGE } from "../../common/links";
+
+const StyledPaper = styled(Paper)`
+  border-radius: 1rem;
+`;
+
+const StyledRowContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const StyledTitle = styled.span`
+  display: table-cell;
+  vertical-align: middle;
+  font-size: 1.5em;
+  margin: 0.5em 0;
+`;
+
+interface EditProfileErrors {
+  username: string;
+  email: string;
+}
+
+const DEFAULT_ERRORS: EditProfileErrors = {
+  username: "",
+  email: "",
+};
+
+interface EditProfileState {
+  username: string;
+  email: string;
+}
+
+interface EditProfileDialogProps {
+  open: boolean;
+  onClose: () => any;
+  username: string | undefined;
+  email: string | undefined;
+}
+
+const EditProfileDialog = ({ open, onClose }: EditProfileDialogProps) => {
+  const userState = useAppSelector((state) => state.user);
+  let curUsername = userState.username;
+  let curEmail = userState.email;
+
+  const DEFAULT_FIELDS = {
+    username: curUsername ? curUsername : "",
+    email: curEmail ? curEmail : "",
+  };
+
+  const [values, setValues] = useState<EditProfileState>(DEFAULT_FIELDS);
+  const [errors, setErrors] = useState<EditProfileErrors>(DEFAULT_ERRORS);
+  const dispatch = useAppDispatch();
+
+  let history = useHistory();
+
+  const handleClose = () => {
+    setValues(DEFAULT_FIELDS);
+    setErrors(DEFAULT_ERRORS);
+    onClose();
+  };
+
+  const handleUpdateUser = async () => {
+    try {
+      const { username, email } = values;
+      const validationResult = await userService.validate({
+        username: username,
+        email: email,
+      });
+      let isValid = true;
+      let formErrors = errors;
+
+      // TODO: check if existing email belongs to this user, if not ensure updated email isn't being used
+      if (email !== curEmail && validationResult.emailExists) {
+        formErrors = { ...formErrors, email: "The new email already exists" };
+        isValid = false;
+      }
+
+      if (validationResult.usernameExists) {
+        formErrors = { ...formErrors, username: "The username already exists" };
+        isValid = false;
+      }
+
+      if (!isValid) {
+        setErrors(formErrors);
+      } else {
+        if (userState._id) {
+          const dispatchedAction = await dispatch(
+            updateUser({
+              _id: userState._id,
+              username: username,
+              email: email,
+            }),
+          );
+          unwrapResult(dispatchedAction);
```

Comment:
just double checking, is this necessary?

Reply:
Probably not

Reply:
I think it is because I found if I didn't have it then it wouldn't bubble up the errors in the response.